internal/blockchain/ethereum/ethereum.go

```diff
@@ -878,17 +878,47 @@ func (e *Ethereum) AddContractListener(ctx context.Context, listener *core.Contr
 			return err
 		}
 	}
-	abi, err := ffi2abi.ConvertFFIEventDefinitionToABI(ctx, &listener.Event.FFIEventDefinition)
-	if err != nil {
-		return i18n.WrapError(ctx, err, coremsgs.MsgContractParamInvalid)
+
+	filters := make([]*filter, 0)
+	// This keeps the existing behaviour where a contract listener could only listen to one event
+	if listener.Event != nil {
```

Comment:
Yeah makes sense - I think this was done before for reload of contract listeners on restart, but how not standardise that on load of listeners from DB instead of here!

Reply:
Need to address this now